qemu/tests/x86_cpu_asyncpf.py

```diff
@@ -0,0 +1,82 @@
+import re
+import logging
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_qemu
+
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Enable interrupt based asynchronous page fault mechanism by default
+
+    1. boot a guest with "-machine pc-q35-rhel8.5.0" or newer.
+    2. check guest's dmesg, should see Async PF enabled
+    3. check 'Hypervisor callback interrupts' inside guest,
+       should have at least 1 for each CPU.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def boot_and_check_guest(machine_type):
+        """
+        Boot guest and check async PF inside guest
+
+        :param machine_type: Boot guest with which machine type
+        """
+        params['machine_type'] = machine_type
+        check_async_PF = params['check_async_PF']
+        check_interrupts = params['check_interrupts']
```

Comment:
Let's put it out of this function, so we don't need get it multiple times.

Reply:
> Let's put it out of this function, so we don't need get it multiple times.

We only use it in this function, and only get it once.

Reply:
You mean this function will only be called once?  Then it's ok.

Reply:
> You mean this function will only be called once? Then it's ok.

yes.